src/Orders.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from "react";
+import RestaurantButton from './RestaurantButton'
+
+class Orders extends Component {
+    constructor(props) {
+        super(props);
+        this.state = { orders: 0 }
+    }
+
+    addOrder = () => {
+        const previousCount = this.state.orders;
+        this.setState({ orders: previousCount + 1 });
```

Comment:
This isn't quite right. It will work _most_ of the time but if you click very quickly on the button you may notice that it will occasionally miss incrementing the number for a click.

This is because React will sometimes _batch_ multiple calls to `this.setState()` (this means that it will combine them together). This means that `previousCount` here will be wrong. To get around this, you must use the callback version of `setState`:

```js
this.setState((previousState) => {
  // ...
})
```

Can you fill out the rest?

The rule of thumb about which version of `setState` to use is that if you need to calculate the new state based from the previous state, then you need to use the callback version. If you don't need that then the simple object version is fine.